tasks.mak

```diff
@@ -31,61 +33,102 @@ install_tools:
 # Build
 ############
 
+.PHONY: build
 build:
-	mkdir -p /tmp/google-cloud-ops-agent	
+	mkdir -p /tmp/google-cloud-ops-agent
 	DOCKER_BUILDKIT=1 docker build -o /tmp/google-cloud-ops-agent . $(ARGS)
 
+.PHONY: fluent_bit_local
+fluent_bit_local:
+	mkdir -p $(PWD)/dist
+	$(shell ./builds/fluent_bit_local.sh $(PWD)/dist)
+
+.PHONY: otelopscol_local
+otelopscol_local:
+	mkdir -p $(PWD)/dist
+	$(shell ./builds/otel_local.sh $(PWD)/dist)
+
 ############
 # Tools
 ############
 
+.PHONY: addlicense
 addlicense:
-	addlicense -ignore "submodules/**" -ignore "**/testdata/**" -ignore "**/built-in-config*" -c "Google LLC" -l apache . 
+	addlicense -ignore "submodules/**" -ignore "**/testdata/**" -ignore "**/built-in-config*" -c "Google LLC" -l apache .
 
+.PHONY: addlicense_check
 addlicense_check:
-	addlicense -check -ignore "submodules/**" -ignore "**/testdata/**" -ignore "**/built-in-config*" -c "Google LLC" -l apache . 
+	addlicense -check -ignore "submodules/**" -ignore "**/testdata/**" -ignore "**/built-in-config*" -c "Google LLC" -l apache .
 
-# Originally I made the argument to this "PATH" but of course that messes with
-# the PATH environment variable. :D
+.PHONY: addlicense_to
 addlicense_to:
 	addlicense -c "Google LLC" -l apache $(P)
 
+.PHONY: yaml_format
 yaml_format:
 	yamlfmt
 
+.PHONY: yaml_lint
 yaml_lint:
 	yamlfmt -lint
 
+.PHONY: compile_dockerfile
 compile_dockerfile:
 	go run ./dockerfiles
 
 ############
 # Unit Tests
 ############
 
+.PHONY: test
 test:
 	go test ./...
 
+.PHONY: test_confgenerator
+test_confgenerator:
+	go test ./confgenerator $(if $(UPDATE),-update,)
+
+.PHONY: test_confgenerator_update
 test_confgenerator_update:
-	go test ./confgenerator -update
+	UPDATE=true $(MAKE) test_confgenerator
 
+.PHONY: test_metadata
+test_metadata:
+	go test ./integration_test/metadata $(if $(UPDATE),-update,)
+
+.PHONY: test_metadata_update
 test_metadata_update:
-	go test ./integration_test/metadata -update
+	UPDATE=true $(MAKE) test_metadata
 
+.PHONY: new_confgenerator_test
 new_confgenerator_test:
 ifndef TEST_NAME
 	$(error "Please provide a TEST_NAME argument")
 endif
-	mkdir -p ./confgenerator/testdata/goldens/$(TEST_NAME) 
+	mkdir -p ./confgenerator/testdata/goldens/$(TEST_NAME)
 	mkdir -p ./confgenerator/testdata/goldens/$(TEST_NAME)/golden
 	touch ./confgenerator/testdata/goldens/$(TEST_NAME)/input.yaml
 
+.PHONY: transformation_test
+transformation_test:
+ifeq ("$(wildcard $(PWD)/dist/fluent-bit)","")
+	$(MAKE) fluent_bit_local
+endif
+ifeq ("$(wildcard $(PWD)/dist/otelopscol)","")
+	$(MAKE) otelopscol_local
+endif
```

Comment:
Can we just use make dependencies for this? E.g.:
```make
dist/fluent-bit:
	$(MAKE) fluent_bit_local
dist/otelopscol:
	$(MAKE) otelopscol_local
…
transformation_test: dist/fluent_bit dist/otelopscol
	FLB=… OTELOPSCOL=… go test …
```

Reply:
Yep good idea, done in [73e5155](https://github.com/GoogleCloudPlatform/ops-agent/pull/1722/commits/73e5155e2c42e0bbdb8fbc2490456103a33f822a)